ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -49,65 +53,79 @@ import kotlin.collections.component2
 import kotlin.collections.set
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.reactive.asFlow
 
 /** View model for the Data Collection fragment. */
-@OptIn(ExperimentalCoroutinesApi::class)
 @HiltViewModel
 class DataCollectionViewModel
 @Inject
 internal constructor(
   private val viewModelFactory: ViewModelFactory,
-  private val submissionRepository: SubmissionRepository,
-  private val locationOfInterestRepository: LocationOfInterestRepository,
   private val locationOfInterestHelper: LocationOfInterestHelper,
   private val popups: Provider<EphemeralPopups>,
   private val navigator: Navigator,
+  private val submitDataUseCase: SubmitDataUseCase,
   @ApplicationScope private val externalScope: CoroutineScope,
   @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
   private val savedStateHandle: SavedStateHandle,
+  private val resources: Resources,
+  locationOfInterestRepository: LocationOfInterestRepository,
   surveyRepository: SurveyRepository,
 ) : AbstractViewModel() {
 
-  // TODO(#1541): Set loiId once Suggest LOI task is completed.
-  private val loiId: StateFlow<String?> = MutableStateFlow(savedStateHandle["locationOfInterestId"])
+  private val loiId: String? = savedStateHandle["locationOfInterestId"]
   private val activeSurvey: Survey = requireNotNull(surveyRepository.activeSurvey)
   private val job: Job =
     activeSurvey.getJob(requireNotNull(savedStateHandle["jobId"])).orElseThrow()
+  val tasks: List<Task> = buildList {
+    when (val suggestTaskType = job.suggestLoiTaskType) {
+      Task.Type.DROP_A_PIN ->
+        add(
+          Task(
+            id = "-1",
+            index = -1,
+            suggestTaskType,
+            resources.getString(R.string.new_location),
+            isRequired = true
+          )
+        )
+      Task.Type.DRAW_POLYGON ->
+        add(
+          Task(
+            id = "-1",
+            index = -1,
+            suggestTaskType,
+            resources.getString(R.string.new_location),
+            isRequired = true
+          )
+        )
+      else -> {}
+    }
+    addAll(job.tasksSorted)
```

Comment:
This part is not clear to me. Can you please consider creating a function with appropriate name for clarity?

Reply:
Okay I think I was able to simplify this a fair bit and pulled out the shared logic into a createSuggestLoiTask function.